source/unwind.js

```diff
@@ -0,0 +1,54 @@
+import _curry2 from './internal/_curry2.js';
+import has from './has.js';
+import propIs from './propIs.js';
+import reduce from './reduce.js';
+import clone from './clone.js';
+import dissoc from './dissoc.js';
+import assoc from './assoc.js';
+
+/**
+ *
+ * Deconstructs an array field from the input documents to output a document for each element.
+ * Each output document is the input document with the value of the array field replaced by the element.
+ *
+ * @func
+ * @memberOf R
+ * @category Object
+ * @sig String -> {k: [v]} -> [{k: v}]
+ * @param {String} key The key to determine which property of the object should be unwind
+ * @param {Object} object The object containing list under property named as key which is to unwind
+ * @return {List} A new list of object containing the value of input key having list replaced by each element in the object.
+ * @example
+ *
+ *  R.unwind('hobbies', {
+ *  name: 'alice',
+ *  hobbies: ['Golf', 'Hacking'],
+ *  colors: ['red', 'green'],
+ * });
```

Comment:
Perhaps a stupid comment: would this work if one attempted to lift `unwind` so as to produce:

```
[ alice x golf x red
, alice x golf x green
, alice x hacking x red
, alice x hacking x green
]
```

Reply:
It would be easy enough to add a wrapper function:

```js
const unwindMany = (keys) => (obj) => 
  keys.reduce((a, p) => a .flatMap (unwind (p)), [obj])

const object = { 
  name: 'alice',
  hobbies: ['Golf', 'Hacking'],
  colors: ['red', 'green']
}

unwindMany(['hobbies', 'colors'])(object) //=> [
//   {colors: "red", hobbies: "Golf", name: "alice"},
//   {colors: "green", hobbies: "Golf", name: "alice"}, 
//   {colors: "red", hobbies: "Hacking", name: "alice"}, 
//   {colors: "green", hobbies: "Hacking", name: "alice"}
// ]
```

Or perhaps:

```js
const unwindMany = (keys) => (obj) => 
  reduce ((a, p) => chain (unwind (p)) (a), [obj], keys)
```

(Or some totally point-free version if so desired.)

But I don't see how to do something like this with `lift`.

Reply:
This wrapper function looks useful. Should `unwind` attempt to support a list of keys?

Reply:
> This wrapper function looks useful. Should unwind attempt to support a list of keys?

See #3078, where I suggested all sorts of extensions to this, but we eventually decided that simplicity should rule.  This also included a simplification from the original suggestion, which would have made `unwind` more like `map(<the unwind above>)`.  That makes the wrapper uglier than I would like, but I think is still the right call.  (It would then have been something like `const unwindMany = reduce (unwind)`, which is quite nice.)  If there is a call for it, we can later simply add some version of `unwindMany`.